packages/core-components/src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -21,16 +21,53 @@ import { useTheme } from '@material-ui/core';
 import { BackstageTheme } from '@backstage/theme';
 import { CopyTextButton } from '../CopyTextButton';
 
-type Props = {
+/**
+ * Properties for {@link CodeSnippet}
+ */
+export interface CodeSnippetProps {
+  /**
+   * Code Snippet text
+   */
   text: string;
+  /**
+   * Language used by {@link .text}
+   */
   language: string;
+  /**
+   * Whether to show line number
+   *
+   * @remarks
+   *
+   * Default: false
+   */
   showLineNumbers?: boolean;
+  /**
+   * Whether to show button to copy code snippet
+   *
+   * @remarks
+   *
+   * Default: false
```

Comment:
I couldn't make https://api-extractor.com/pages/tsdoc/tag_defaultvalue/ work.
I think support is half-baked as RFC linked on page is still open